app/core/model/src/main/java/io/syndesis/model/integration/IntegrationDeployment.java

```diff
@@ -36,388 +38,434 @@
-        return integrationId;
+    @Value.Default
+    default IntegrationDeploymentState getCurrentState() {
+        return IntegrationDeploymentState.Pending;
     }
 
-    @JsonProperty("version")
-    public Optional<Integer> getVersion() {
-        return version;
+    @Value.Default
+    default IntegrationDeploymentState getTargetState() {
+        return IntegrationDeploymentState.Active;
     }
 
-    @Override
-    @JsonProperty("name")
-    public String getName() {
-        return name;
+    @Value.Default
+    default Date getLastUpdated() {
+        return new Date(0);
     }
 
-
-    @JsonProperty("spec")
-    public IntegrationDeploymentSpec getSpec() {
-        return spec;
+    @Value.Default
+    default Date getCreatedDate() {
+        return new Date(0);
     }
 
-    @JsonProperty("stepsDone")
-    public List<String> getStepsDone() {
-        return stepsDone;
+    @Value.Default
+    default List<String> getStepsDone() {
+        return Collections.emptyList();
     }
 
-    @JsonProperty("currentState")
-    public IntegrationDeploymentState getCurrentState() {
-        return currentState;
+    default Optional<String> getIntegrationId() {
+        return getSpec().getId();
     }
 
-    @JsonProperty("targetState")
-    public IntegrationDeploymentState getTargetState() {
-        return targetState;
+    Integration getSpec();
+
+    class Builder extends ImmutableIntegrationDeployment.Builder {
+
     }
 
-    @JsonProperty("timesUsed")
-    public BigInteger getTimesUsed() {
-        return timesUsed;
+    default Builder builder() {
+        return new Builder().createFrom(this);
     }
 
-    @JsonProperty("lastUpdated")
-    @SuppressFBWarnings(value="EI_EXPOSE_REP")
-    public Date getLastUpdated() {
-        return lastUpdated;
+    default IntegrationDeployment withCurrentState(IntegrationDeploymentState state) {
+        return builder().currentState(state).build();
     }
 
-    @JsonProperty("createdDate")
-    @SuppressFBWarnings(value="EI_EXPOSE_REP")
-    public Date getCreatedDate() {
-        return createdDate;
+    default IntegrationDeployment withTargetState(IntegrationDeploymentState state) {
+        return builder().targetState(state).build();
     }
 
-    /**
-     * Returns that {@link IntegrationDeploymentState}.
-     * The state is either the `desired state` or `pending`.
-     * @return true, if current state is matching with target, false otherwise.
-     */
     @JsonIgnore
-    public boolean isPending() {
-        return getTargetState() != getCurrentState();
+    default String getName() {
+        return getSpec().getName();
     }
 
-
     @JsonIgnore
-    public boolean isInactive() {
+    default boolean isInactive() {
         return getCurrentState() == IntegrationDeploymentState.Undeployed || getCurrentState() == IntegrationDeploymentState.Inactive
-            ? true
-            : false;
+                ? true
+                : false;
```

Comment:
do we need the ternary here?

Reply:
Was existing code.  I’d have review closer.

Reply:
right. dont need it.